### ISSUE TITLE
AssertionError  assert object_type == "PRIVATE KEY"             

### ISSUE THREAD

**Issue by nmweizi** [saleemrashid/badecparams#4]
using curve name prime256v1 instead of secp256r1                                            
Writing CERTIFICATE to 'intermediateCA.crt'                                                 
Writing CERTIFICATE to 'intermediateCA.crt'                                                 
Writing EC PRIVATE KEY to 'intermediateCA.key'                                              
Generating a 4096 bit RSA private key                                                       
...............++                                                                           
...................................................................................++       
writing new private key to '-'                                                              
-----                                                                                       
Traceback (most recent call last):                                                          
  File "badecparams.py", line 376, in <module>                                              
    main()                                                                                  
  File "badecparams.py", line 350, in main                                                  
    "organizational_unit_name": get_name("Code Signing Authority"),                         
  File "badecparams.py", line 233, in write_authenticode_certificate                        
    private_key, public_key = generate_private_key("rsa:4096")                              
  File "badecparams.py", line 143, in generate_private_key                                  
    assert object_type == "PRIVATE KEY"                                                     
AssertionError

**Comment by nmweizi** [comment]
update openssl

**Comment by hktalent** [comment]
Mac OS, python3, try
```
brew upgrade openssl
```
but,me too
```
Generating a RSA private key
...........................................++++
..........................................................++++
writing new private key to '-'
-----
Traceback (most recent call last):
  File "./badecparams.py", line 381, in <module>
    main()
  File "./badecparams.py", line 355, in main
    "organizational_unit_name": get_name("Code Signing Authority"),
  File "./badecparams.py", line 238, in write_authenticode_certificate
    private_key, public_key = generate_private_key("rsa:4096")
  File "./badecparams.py", line 148, in generate_private_key
    assert object_type == "PRIVATE KEY"
AssertionError
```

**Comment by saleemrashid** [comment]
This is because `openssl` is OpenSSL 1.0 on your system, instead of OpenSSL 1.1. Fixed in d23c49447163e17a28d9d72365cf9039161b6542.
